planning/behavior_velocity_planner/src/scene_module/occlusion_spot/occlusion_spot_utils.cpp

```diff
@@ -228,244 +224,210 @@ void calcSlowDownPointsForPossibleCollision(
+std::vector<PredictedObject> getParkedVehicles(
+  const PredictedObjects & dyn_objects, const PlannerParam & param,
+  std::vector<Point> & debug_point)
 {
-  autoware_auto_planning_msgs::msg::Path path;
-  for (const auto & p : path_with_id.points) {
-    path.points.push_back(p.point);
+  std::vector<PredictedObject> parked_vehicles;
+  std::vector<Point> points;
+  for (const auto & obj : dyn_objects.objects) {
+    bool is_parked_vehicle = true;
+    if (!occlusion_spot_utils::isStuckVehicle(obj, param.stuck_vehicle_vel)) {
+      continue;
+    }
+    const geometry_msgs::msg::Point & p = obj.kinematics.initial_pose_with_covariance.pose.position;
+    BasicPoint2d obj_point(p.x, p.y);
+    if (is_parked_vehicle) {
+      parked_vehicles.emplace_back(obj);
+      points.emplace_back(p);
+    }
   }
-  return path;
+  debug_point = points;
+  return parked_vehicles;
 }
 
-lanelet::ConstLanelet buildPathLanelet(
-  const autoware_auto_planning_msgs::msg::PathWithLaneId & path)
+ArcCoordinates getOcclusionPoint(const PredictedObject & obj, const ConstLineString2d & ll_string)
 {
-  autoware_auto_planning_msgs::msg::Path converted_path = filterLitterPathPoint(toPath(path));
-  if (converted_path.points.empty()) {
-    return lanelet::ConstLanelet();
-  }
-  // interpolation is done previously
-  lanelet::BasicLineString2d path_line;
-  path_line.reserve(converted_path.points.size());
-  // skip last point that will creates extra ordinary path
-  for (size_t i = 0; i < converted_path.points.size() - 1; ++i) {
-    const auto & pose_i = converted_path.points[i].pose;
-    path_line.emplace_back(pose_i.position.x, pose_i.position.y);
+  Polygon2d poly = planning_utils::toFootprintPolygon(obj);
+  std::deque<lanelet::ArcCoordinates> arcs;
+  for (const auto & p : poly.outer()) {
+    lanelet::BasicPoint2d obj_p = {p.x(), p.y()};
+    arcs.emplace_back(lanelet::geometry::toArcCoordinates(ll_string, obj_p));
   }
-  // set simplified line to lanelet
-  lanelet::BasicLineString2d simplified_line;
-  boost::geometry::simplify(path_line, simplified_line, 0.1);
-  lanelet::Points3d path_points;
-  path_points.reserve(simplified_line.size());
-  for (const auto & p : simplified_line) {
-    path_points.emplace_back(lanelet::InvalId, p.x(), p.y(), 0.0);
+  /** remove
+   *  x--------*
+   *  |        |
+   *  x--------* <- return
+   * Ego===============> path
+   **/
+  // sort by arc length
+  std::sort(arcs.begin(), arcs.end(), [](ArcCoordinates arc1, ArcCoordinates arc2) {
+    return arc1.length < arc2.length;
+  });
+  // remove closest 2 polygon point
+  arcs.pop_front();
+  arcs.pop_front();
+  // sort by arc distance
+  std::sort(arcs.begin(), arcs.end(), [](ArcCoordinates arc1, ArcCoordinates arc2) {
+    return std::abs(arc1.distance) < std::abs(arc2.distance);
+  });
+  // return closest to path point which is choosen by farthest 2 points.
+  return arcs.at(0);
+}
+
+double calcSgnValue(const double val, const double offset)
```

Comment:
Can this function have a more descriptive name?

Reply:
calculateOffsetValue is good?